src/SyntaxHighlighter/Cpp.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "SyntaxHighlighter/Cpp.h"
+
+#include <QColor>
+#include <QRegularExpression>
+#include <QString>
+#include <QTextCharFormat>
+
+namespace orbit_syntax_highlighter {
+namespace C {
+namespace {
+enum HighlighterState { kInitialState, kOpenCommentState, kOpenStringState };
+
+const QColor kGrey{0x99, 0x99, 0x99};
+const QColor kBlue{0x61, 0x96, 0xcc};
+const QColor kYellow{0xf0, 0xf0, 0x55};
+const QColor kYellowOrange{0xf8, 0xc5, 0x55};
+const QColor kOrange{0xf0, 0x8d, 0x49};
+const QColor kOlive{0x80, 0x80, 0x00};
+const QColor kGreen{0x7e, 0xc6, 0x99};
+const QColor kViolet{0xcc, 0x99, 0xcd};
+
+const char* const kNumberRegex =
+    "(?:\\b0x(?:[\\da-f]+(?:\\.[\\da-f]*)?|\\.[\\da-f]+)(?:p[+-]?\\d+)?|(?:\\b\\d+(?:\\.\\d*)?|"
+    "\\B\\.\\d+)(?:e[+-]?\\d+)?)[ful]{0,4}";
+const char* const kConstantRegex =
+    "__FILE__|__LINE__|__DATE__|__TIME__|__TIMESTAMP__|__func__|EOF|NULL|SEEK_CUR|SEEK_END|SEEK_"
+    "SET|stdin|stdout|stderr";
+const char* const kKeywordRegex =
+    "\\b(?:__attribute__|_Alignas|_Alignof|_Atomic|_Bool|_Complex|_Generic|_Imaginary|_Noreturn|_"
+    "Static_assert|_Thread_local|asm|typeof|inline|auto|break|case|char|const|continue|default|do|"
+    "double|else|enum|extern|float|for|goto|if|int|long|register|return|short|signed|sizeof|static|"
+    "struct|switch|typedef|union|unsigned|void|volatile|while|)\\b";
+const char* const kCommentRegex =
+    "\\/\\/(?:[^\\r\\n\\\\]|\\\\(?:\\r\\n?|\\n|(?![\\r\\n])))*|\\/\\*[\\s\\S]*?\\*\\/";
+const char* const kOpenCommentRegex = "\\/\\*([^\\*]|[\\*]+[^\\/])*?$";  // Don't work with **/
+const char* const kEndCommentRegex = "[\\s\\S]*\\*\\/";
+const char* const kFunctionRegex = "[a-z_]\\w*(?=\\s*\\()";
+const char* const kNoEndCommentRegex = "([^\\*]|\\*+[^\\/\\*])*$";
+const char* const kMacroRegex = "(^\\s*)#\\s*[a-z_]\\S*";
```

Comment:
Should this better be called kPreprocessorRegex as you are detecting preprocessor directives and not macros (which may be indistinguishable from functions in general)?